### ISSUE TITLE
Make user.email column unique

### ISSUE THREAD

**Issue by adamalton** [web3-storage/web3.storage#1388]
Currently, the `email` column in the `public.user` table is not unique. On production we have at least one email address which appears twice in the table. It would seem logical to me that this column should have a unique constraint on it, but perhaps there is a reason why it doesn't?

Perhaps this occurs when a user authenticates with their email address using magic.link and then separately authenticates using their GitHub account which happens to have the same email address. And perhaps we don't trust that GitHub has validated their email address and therefore don't trust that they are the same person? If so, is there a way around that?

If we want to continue with the premise that the `email` column is not guaranteed to be unique, then we should probably remove the `getUserByEmail` method from the `DBClient` class, and replace it with something more admin-user-specific for the current use case of getting the support@web3.storage admin user.

If we add a constraint, it should probably be case-insensitive.

**Comment by vasco-santos** [comment]
>  It would seem logical to me that this column should have a unique constraint on it, but perhaps there is a reason why it doesn't

Magic link handles differently a github auth and email auth unfortunately. This means that the public address it creates is different even if user has same email in github account.

It would be great to have a way of unifying these accounts. Probably we should wait on the new auth setup

**Comment by dchoi27** [comment]
yeah let's mark this blocked for now bc of UCANification coming. agreed it's annoying
